src/scripts/Update.ts

```diff
@@ -598,6 +599,49 @@ class Update implements Saveable {
             saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 32, 34); // New Mauville
             saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 84, 64); // Pledge Grove
             saveData.statistics.dungeonsCleared = Update.moveIndex(saveData.statistics.dungeonsCleared, 78, 79); // Abundant Shrine
+
+            // Shards -> Gems
+            saveData.quests.questList = {
+                ...saveData.quests.questList,
+                gainGemsQuest: saveData.quests.questList.gainShardsQuest || 0,
+            };
```

Comment:
The `questList` is an array of quest, each with a structure like the following:
```json
{
  "index": 0,
  "data": [
    319,
    1250,
    10
  ],
  "initial": 208068,
  "claimed": true,
  "notified": true,
  "name": "GainShardsQuest"
}
```
You would basically just need to loop through it and replace any name instance of `GainShardsQuest` with `GainGemsQuest`

Something like this _should_ work, I haven't tested it though:
```js
saveData.quests.questList = saveData.quests.questList.map(q => {
  if (q.name == 'GainShardsQuest') q.name = 'GainGemsQuest';
  return q;
})
```

Reply:
Works great!